### ISSUE TITLE
Detect redundant calls to str

### ISSUE THREAD

**Issue by borkdude** [clj-kondo/clj-kondo#2323]
- `(str "foo")`
- `(str (format "foo"))`

See branch `redundant-call-string`

**Comment by borkdude** [comment]
One false positive:

```
(are [x y] (= x (str y))
  "foo" "bar"
  "foo" 1)
```

**Comment by NoahTheDuke** [comment]
Dang, nice that you can reuse the "return type" of functions to check when `str` is redundant. I did this in splint but had to hardcode a couple built-in functions.

**Comment by borkdude** [comment]
I like that you split it into a separate linter called `:redundant-str-call`, I might do that too and enable it by default since (perhaps in a later release as to not cause trouble for people with a new linter that might not be perfect yet). Where did you have to hardcode it?

**Comment by NoahTheDuke** [comment]
https://github.com/NoahTheDuke/splint/blob/0dd94d487b9ea9426d3b624aa294c6ee3fa7a1f9/src/noahtheduke/splint/rules/lint/redundant_str_call.clj#L16-L18

I'm checking `(str (FUNC ...))` where `FUNC` is `str` or `format`. I don't know if it's worth checking lots more, I'm still determining where the line is for this kind of check. Once I get to 3+ functions, makes me feel like I should invest in tracking var definitions/locals, which is an incredible amount of work (as seen by clj-kondo).
